### ISSUE TITLE
`~/.kube/config` for Docker Desktop kubernetes needs to be usable from docker

### ISSUE THREAD

**Issue by glyn** [docker/for-mac#3708]
This issue was encountered when executing an image on Docker Desktop which attempts to access the Docker Desktop kubernetes cluster using the host's `~/.kube/config`. The attempt fails because `localhost` has a different meaning inside Docker than it does in the host.

  - [X] I have tried with the latest version of my channel (Stable or Edge)
  - [X] I have uploaded Diagnostics
  - Diagnostics ID: 4A70697A-9F32-4AEC-9B7D-F16E8AE5FE28/20190611132519

### Expected behavior

The Docker Desktop kubernetes cluster can be accessed from Docker using the information in `~/.kube/config`.

### Actual behavior

The cluster cannot be accessed from Docker and the following message is logged:
```
The connection to the server localhost:6443 was refused - did you specify the right host or port?
```

### Information
  - Docker Desktop for macOS edge version: 2.0.4.1 (34207)
  - macOS version: 10.14.5

### Diagnostic logs

Uploaded - see above.

### Steps to reproduce the behavior

1. Install docker desktop and enable kubernetes. Ensure it is selected for kubectl.
2. Build `duffle` at `master` by cloning https://github.com/deislabs/duffle and following the instructions in the [developers guide](https://github.com/deislabs/duffle/blob/master/docs/developing.md).
3. Set up `duffle` by issuing `duffle init`.
4. Use the following example: https://github.com/glyn/duffle/tree/hello-k8s/examples/hellok8s
5. Build the example:
    ```
    duffle build ./examples/hellok8s
    ```
6. Create a credential pointing to `~/.kube/config`:
    ```    
    duffle credentials add examples/hellok8s/myk8s.yaml
    ```
    where `examples/hellok8s/myk8s.yaml` has the following contents:
    ```
    name: myk8s
    credentials:
      - name: kubeconfig
        source:
          path: $HOME/.kube/config
    ```
7. Attempt to install the example and observe the error:
    ```
    duffle install hellok8s-demo hellok8s:0.1 --credentials myk8s
    Executing install action...
    The connection to the server localhost:6443 was refused - did you specify the right host or port?
    Action install complete for hellok8s-demo
    ```

### Background

See https://github.com/deislabs/duffle/issues/762 for the context.

**Comment by silvin-lubecki** [comment]
Thank you @glyn for opening this issue 👍

**Comment by silvin-lubecki** [comment]
This bug is fixed with the latest 2.1.0.0 Edge version of Docker Desktop.
